scripts/pyi_generator.py

```diff
@@ -638,27 +687,44 @@ def _scan_file(self, module_path: Path):
         )
         self._write_pyi_file(module_path, ast.unparse(new_tree))
 
-    def _scan_files_multiprocess(self, files):
+    def _scan_files_multiprocess(self, files: list[Path]):
         with Pool(processes=cpu_count()) as pool:
             pool.map(self._scan_file, files)
 
-    def scan_all(self, targets):
+    def scan_all(self, targets, changed_files: list[Path] | None = None):
         """Scan all targets for class inheriting Component and generate the .pyi files.
 
         Args:
             targets: the list of file/folders to scan.
+            changed_files (optional): the list of changed files since the last run.
         """
         file_targets = []
         for target in targets:
-            path = Path(target)
-            if target.endswith(".py") and path.is_file():
-                file_targets.append(path)
-            elif path.is_dir():
-                for root, _, files in os.walk(path):
+            target_path = Path(target)
+            if target_path.is_file() and target_path.suffix == ".py":
+                file_targets.append(target_path)
+            elif target_path.is_dir():
+                for root, _, files in os.walk(target_path):
```

Comment:
https://docs.python.org/3/library/pathlib.html#pathlib.Path.walk

Reply:
yeah, i'm slowly getting the whole team on board with the new APIs, we currently have a mixture in use, depending on which file you're in.

Reply:
sadly, `Path.walk` is only available since python 3.12. However, I still migrated this line to pathlib using a different approach